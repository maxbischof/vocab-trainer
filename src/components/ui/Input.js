import React from 'react'
import styled from 'styled-components'

const Input = React.forwardRef(
  ({ name, type, placeholder, onChange, value, error }) => (
    <StyledInput
      name={name}
      type={type}
      placeholder={placeholder}
      onChange={onChange}
      value={value}
      error={error}
    ></StyledInput>
  )
)

const StyledInput = styled.input`
  border: none;
  border-bottom: 1px solid var(--text);
  border-radius: 0;
  padding: 5px;
  margin: 10px 0 10px 0;
  ${(props) =>
    props.error &&
    props.error === props.name &&
    'border: 2px solid var(--error)'}
`

export default Input
