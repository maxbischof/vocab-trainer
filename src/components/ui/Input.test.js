import React from 'react'
import Input from './Input'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<Input name="word" type="text" placeholder="Type here" value="" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
