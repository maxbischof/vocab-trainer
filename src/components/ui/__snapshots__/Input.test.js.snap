// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="sc-bdfBwQ kCvuAu"
  name="word"
  placeholder="Type here"
  type="text"
  value=""
/>
`;

exports[`renders correctly with error 1`] = `
<input
  className="sc-bdfBwQ kFiDjn"
  name="word"
  placeholder="Type here"
  type="text"
  value=""
/>
`;
